src/System.Runtime/tests/System/StringGetHashCodeTests.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Xunit;
+
+namespace System.Tests
+{
+    public class StringGetHashCodeTests : RemoteExecutorTestBase
+    {
+        [Fact]
+        /// <summary>
+        /// Ensure that string hash codes are randomized by getting the hash code for the same string in two processes
+        /// and confirming it is different (modulo possible values of int).
+        /// If the legacy hash codes are being returned, it will not be different.
+        /// </summary>
+        public void GetHashCode_UseSameStringInTwoProcesses_ReturnsDifferentHashCodes()
+        {
+            const string abc = "same string in different processes";
+            int childHashCode, parentHashCode = abc.GetHashCode(), timesTried = 0;
+
+            Func<int> GetChildHashCode = () =>
+            {
+                using (RemoteInvokeHandle handle = RemoteInvoke(() => abc.GetHashCode(), new RemoteInvokeOptions { CheckExitCode = false }))
+                {
+                    handle.Process.WaitForExit();
+                    return handle.Process.ExitCode;
```

Comment:
did you verify (eg with a couple console.writelines) that the hash code is getitng successfully transmitted? If it was getting cropped down to two bytes or something, this test would still pass...

Reply:
yes. I just verified. it's getting passed down as expected

Reply:
great. a nice clean set of tests.